### PR TITLE
DRIVERS-2732 CSFLE/QE KMIP support for encrypt/decrypt

### PR DESCRIPTION
**Pull request by adriandole**
Please complete the following before merging:

- [x] Update changelog.
- [x] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver. [C implementation](https://github.com/mongodb/mongo-c-driver/pull/1531/files)
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

File: source/client-side-encryption/tests/legacy/kmipKMS.yml [hunk 1]
```diff
@@ -4,8 +4,34 @@ database_name: &database_name "default"
 collection_name: &collection_name "default"
```
Comment (kevinAlbs) [inline source/client-side-encryption/tests/legacy/kmipKMS.yml:5]: The legacy tests are generated by templates. Suggest adding a kmipKMS.yml.template. That may resolve the failing GitHub action.

File: source/client-side-encryption/client-side-encryption.rst [hunk 2]
```diff
@@ -563,6 +563,7 @@ accept arbitrary strings at runtime for forward-compatibility.
 
    interface KMIPKMSOptions {
       endpoint: string;
+      delegated: boolean; // If true (recommended), the KMS provider will perform DEK encryption and decryption. Defaults to false.
```
Comment (kevinAlbs) [inline source/client-side-encryption/client-side-encryption.rst:566]: Can either `KMIPKMSOptions.delegated` or `masterKey.delegated` options be removed? I expect both are not needed for this feature.

`KMIPKMSOptions.delegated` can be true, false, or unset. `masterKey.delegated` can be true, false, or unset. Including both options may require additional testing and documentation to describe which option applies when they have differing values:

  | KMIPKMSOptions.delegated true | KMIPKMSOptions.delegated false | KMIPKMSOptions.delegated unset
-- | -- | -- | --
**masterKey.delegated true** | Delegated. | ? | Delegated.
**masterKey.delegated false** | ? | Not delegated. | Not delegated.
**masterKey.delegated unset** | Delegated. | Not delegated. | Not delegated.

</div></b>

**Review by kevinAlbs — COMMENTED** [review]
Looks good overall. Suggest removing one of the `delegated` options to simplify the interface.

I suggest waiting until corresponding drivers-evergreen-tools changes (for new KMIP key?), libmongocrypt changes, and C driver changes are approved before merging. The reviews for those changes may reveal required test changes.

File: source/client-side-encryption/client-side-encryption.rst (hunk 2 above)
Comment (kevinAlbs) [inline source/client-side-encryption/client-side-encryption.rst:566]:
```suggestion
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with one additional spec update.

**Review by katcharov — APPROVED** [review]
LGTM

**Review by kevinAlbs — CHANGES_REQUESTED** [review]
The client-side-encryption.rst file was converted to markdown. client-side-encryption.rst now has this note:

> This specification has been converted to Markdown and renamed to
  `client-side-encryption.md <client-side-encryption.md>`_.  
>  Use the link above to access the latest version of the specification as the
  current reStructuredText file will no longer be updated.

Prior to merging, apply the changes to the markdown file and revert the reStructuredText changes.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**